paddle/framework/channel_test.cc

```diff
@@ -60,6 +60,32 @@ TEST(Channel, SufficientBufferSizeDoesntBlock) {
   delete ch;
 }
 
+// This test tests that CloseChannel returns a value of zero
+// immediately to all receivers that are trying to receive from the channel.
+TEST(Channel, ReceiverGetsZeroOnClosedChannel) {
```

Comment:
If the purpose of this PR is to add a unit test to ensure that 

1. receiving from a closed buffered channel returns residual values and then zeros values, and
1. receiving from a closed unbuffered channel doesn't block and returns zero.

then this PR looks too simple to ensure the above two invariants.

Reply:
I think one of the axioms was  that : `A receive from a closed channel returns the residual values and then zeros.`
This PR tries to validate that #1 you explained above holds since we aren't explicitly checking for the zero value anywhere else.